src/Training.cpp

```diff
@@ -158,6 +158,69 @@ void Training::dump_training(int game_score, const std::string& out_filename) {
     dump_training(game_score, chunker);
 }
 
+void Training::dump_training_v2(int game_score, OutputChunker& outchunk) {
+    /**
+     * From chunkparser.py
+     *
+     * V2 Format (8604 bytes total)
+     * int32 version (4 bytes)
+     * 1924 float32 probabilities (7696 bytes)
+     * 112 packed bit planes (896 bytes)
+     * uint8 castling us_ooo (1 byte)
+     * uint8 castling us_oo (1 byte)
+     * uint8 castling them_ooo (1 byte)
+     * uint8 castling them_oo (1 byte)
+     * uint8 side_to_move (1 byte)
+     * uint8 rule50_count (1 byte)
+     * uint8 move_count (1 byte)
+     * int8 result (1 byte)
+     * self.v2_struct = struct.Struct('4s7696s896sBBBBBBBb')
+     */
+    static int VERSION = 1;
+
+    std::stringstream out;
+    for (const auto& step : m_data) {
+        // Store the binary version number (4 bytes)
+        out.write(reinterpret_cast<char*>(&VERSION), sizeof(VERSION));
```

Comment:
1. Portability: is it safe to assume identical byte order between client and server? (I guess it is if only x86-64 is relevant.)
2. The use of stringstream: would it work if you instead simply defined a `struct { int32 version; float probabilities[1924]; ... }`, copied all the data into it, and wrote it out as a blob? (Happy to send a followup cleanup PR if you prefer.)

Reply:
1. Good point. Client side should always store data in little endian format. I know howto do this properly in Linux, but not Windows, any ideas?
2. I tried to keep the code as close to V1 as possible, hence the stringstream. One can view `dump_training` and `dump_training_v2` side by side and easily examine the differences.

Reply:
1. Don't know much about Windows programming myself, but I think Windows is always little endian?
2. Understood.